### PR TITLE
Mangaki's version in the footer!

### LINKED ISSUE #432: Current version of Mangaki should be displayed somewhere
For example, in the footer.

### PR DESCRIPTION
**Pull request by RaitoBezarius**
Screenshot of a version

:rocket: with a link to the commit tree !

Fixes #432.

- Supports Git repository.
- Supports Egg packages (from PyPI / `setup.py`).

### CODE REVIEW

**Comment by jilljenn** [comment]
Est-ce que sur le vrai site, on aura une version longue comme ici ou juste 0.4 ? Je préférerais qu'il n'y ait que 0.4.

**Comment by RaitoBezarius** [comment]
Il y aura une version longue et je préfère qu'il y a la version longue, je m'explique: tant qu'on a pas de déploiements de mangaki.fr qui sont toujours bien taggés, faits depuis stable, etc. Les numéros de versions de la production ne m'informent pas sur le code qui tourne sur mangaki.fr ; genre ça pourrait être 0.3 quelque part entre fin de 0.2 et début de 0.4.

On peut en parler à l'oral.

**Comment by RaitoBezarius** [comment]
Title: short version.
Caption: long version.
Link: link to revision.

File: requirements/dev.txt [hunk 1]
```diff
@@ -18,3 +18,4 @@ factory_boy
 coverage
 django_coverage_plugin
 hypothesis
+setuptools-scm
```
Comment (jilljenn) [inline requirements/dev.txt]: Shouldn't this be in `production.txt`?

**Comment by RaitoBezarius** [comment]
Please @jilljenn give it a try again. It'll show only the tagged version or last tag + `.dev{number of commits}` (so that in production, it should never show `0.4.dev20` if it's not tagged properly :)).

File: mangaki/mangaki/settings.py [hunk 2]
```diff
@@ -26,6 +28,24 @@
...
+try:
+    REPO_DIR = os.path.dirname(BASE_DIR)
+    VERSION = get_version(REPO_DIR)
+except:
+    VERSION = None
+
+# Step 2: if we are a nice package.
+try:
+    if not VERSION:
+        VERSION = pkg_resources.require('mangaki')[0].version
```
Comment (jilljenn) [inline mangaki/mangaki/settings.py]: I couldn't try this part. I trust you.

File: mangaki/mangaki/templatetags/version.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+register = template.Library()
+
+
+@register.simple_tag(takes_context=False)
+def mangaki_version():
+    return settings.VERSION.split('+')[0]
+
+@register.simple_tag(takes_context=False)
+def mangaki_revision():
+    try:
```
Comment (jilljenn) [inline mangaki/mangaki/templatetags/version.py]: Could you provide an example?
Like `settings.VERSION = v0.2.devXX+hash.timestamp` will output hash.
Like `settings.VERSION = v0.5.3` will output ''. Is it expected behavior? I feel like I was lucky.

Comment (RaitoBezarius) [reply mangaki/mangaki/templatetags/version.py]: What do you mean by providing an example? Adding `doctests` or documentation?
But yes, a custom tag will give no revision. If a SHA1 hash is available, it'll be used.

Comment (jilljenn) [reply mangaki/mangaki/templatetags/version.py]: Documentation.

Can't you realize? The day we will get to `v10`, the output will be `'1'`. Right? I think there is a problem.

Comment (RaitoBezarius) [reply mangaki/mangaki/templatetags/version.py]: The day we'll get to `v10`, we'll have `VERSION = 'v10.0.3.devXX+ghash.timestamp`
If I split with `+` then take the last part and split by `.`, won't it be alright?
As for the `VERSION`, splitting by `+` and taking the first part will always give the full tag or the dirty tag, right?

Comment (jilljenn) [reply mangaki/mangaki/templatetags/version.py]: Let's say we have tag = 'v10.1'.
split('+'), take last part => 'v10.1'
split('.'), take first part => 'v10'
[:1] => Return '10'
Will this, as a link, work?

Comment (RaitoBezarius) [reply mangaki/mangaki/templatetags/version.py]: Well, you're right, I should ensure that we have at least two items after `split('+')`.

Comment (jilljenn) [reply mangaki/mangaki/templatetags/version.py]: Can you write a test? 😄

Comment (RaitoBezarius) [reply mangaki/mangaki/templatetags/version.py]: I will.

**Pull request merged**